subsys/logging/Kconfig

```diff
@@ -497,26 +497,33 @@ config LOG_BACKEND_NET_SYST_ENABLE
 
 endif # LOG_BACKEND_NET
 
-config LOG_BACKEND_ADSP
-	bool "Enable Intel ADSP backend"
-	depends on SOC_INTEL_APL_ADSP
+config LOG_BACKEND_RB
+	bool "Enable memory ring buffer backend"
 	select RING_BUFFER
 	help
-	  Enable backend in APL ADSP using memory mailbox
+	  Enable backend in memory using ring buffer to keep messages.
 
-if LOG_BACKEND_ADSP
+if LOG_BACKEND_RB
 
-config LOG_BACKEND_ADSP_RINGBUF_BASE
-	hex "Ringbuf base address"
+config LOG_BACKEND_RB_MEM_BASE
+	hex "Ring buffer memory base address"
 	help
 	  Address of the ring buffer in the memory.
 
-config LOG_BACKEND_ADSP_RINGBUF_SIZE
+config LOG_BACKEND_RB_MEM_SIZE
 	hex "Size of the ring buffer"
 	help
-	  Size of the ring buffer.
+	  Size of the ring buffer. Must be multiply of the message slot
+	  size.
```

Comment:
Unrelated to what this PR is trying to do, but would it make sense to rethink this config option and instead call it `LOG_BACKEND_RB_SLOT_COUNT`. That way you don't need to remind the user of using multiples of `LOG_BACKEND_RB_SLOT_SIZE`.

Reply:
`SLOT_COUNT` still need to be multiples of memory size so nothing changes.
There are advantages using `SLOT_SIZE` like it is human understandable size of the memory slot which we can adjust if we see slots are used inefficiently.

Reply:
> SLOT_COUNT still need to be multiples of memory size so nothing changes.

Why? It's not a memory size as-is. You'd get the total memory size by multiplying it with the SLOT_SIZE.

> There are advantages using SLOT_SIZE like it is human understandable size of the memory slot which we can adjust if we see slots are used inefficiently.

That's fair. Though since you use the word "Must" I guess you should back that up by adding a BUILD_ASSERT? Alternatively you could reword this as "*should* be a multiple, or memory will be wasted"

Reply:
We have `BUILD_ASSERT` here: https://github.com/finikorg/zephyr/blob/41f6adb1b56dd118e49f3f43d520483df6c65297/subsys/logging/log_backend_rb.c#L28

>Why? It's not a memory size as-is. You'd get the total memory size by multiplying it with the SLOT_SIZE.

We usually get total available memory window (like https://github.com/finikorg/zephyr/blob/41f6adb1b56dd118e49f3f43d520483df6c65297/soc/xtensa/intel_apl_adsp/Kconfig.defconfig#L55) If we have count * slot size we need to check that it fits to the window.

Reply:
@finikorg thanks for clarifying!